### PR TITLE
Fix VB formatting to stop introducing a space before the ? in a conditional access expression

### LINKED ISSUE #4619: VB Formatting inserts space between > and ? in XML literal followed by conditional access expression
```VB
Class C
    Sub M()
        Dim x = <a/>?.<b>
    End Sub
End Class
```

Expected:

```VB
Class C
    Sub M()
        Dim x = <a/>?.<b>
    End Sub
End Class
```

Actual:

```VB
Class C
    Sub M()
        Dim x = <a/> ?.<b>
    End Sub
End Class
```

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #4619.

The code to handle spaces before the ? in a conditional access expression was very specific, only removing the space if the previous token was of a specific kind. However, there were several cases missed (and one bug after MyClass). Now we take a more general approach that is similar to how we treat member-access expressions: If we see a conditional access expression, we assume that there should never be a space before the ?. Any cases where a space is desired would be exceptional.

### CODE REVIEW

**Comment by DustinCampbell** [comment]
Reviewers: @balajikris, @brettfo, @CyrusNajmabadi, @davkean, @dpoeschl, @jasonmalinowski, @jmarolf, @Pilchie, @rchande

**Comment by Pilchie** [comment]
:+1:

**Comment by balajikris** [comment]
LGTM. :+1:  

Thought it looks like it'd work from the code, do you think you need additional tests for chained conditionals in xml literals and malformed xml literals ?

**Comment by DustinCampbell** [comment]
@balajikris, I don't think additional tests are needed. The code will work when conditional access expressions are chained. Honestly, it just removes the space if a conditional access expression exists at all, regardless of chaining, the tokens/nodes before it, etc.

If there are specific interesting cases, then yes, we should add tests. We don't have any other tests for chaining, so I'll add that. However, spending time on malformed XML literals seem like a rabbit hole with no useful fruit at the bottom.

File: src/Workspaces/VisualBasic/Portable/Formatting/Rules/AdjustSpaceFormattingRule.vb [hunk 1]
```diff
@@ -197,59 +197,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Formatting
...
-            ' MyBase ? [conditional access operator]
-            If previousToken.Kind = SyntaxKind.MyBaseKeyword AndAlso currentToken.Kind = SyntaxKind.QuestionToken AndAlso
-                    currentToken.Parent.Kind = SyntaxKind.ConditionalAccessExpression Then
-
+               previousToken.Parent.IsKind(SyntaxKind.ConditionalAccessExpression) Then
                 Return CreateAdjustSpacesOperation(0, AdjustSpacesOption.ForceSpacesIfOnSingleLine)
             End If
 
-            ' MyClass ? [conditional access operator]
-            If previousToken.Kind = SyntaxKind.MyClassExpression AndAlso currentToken.Kind = SyntaxKind.QuestionToken AndAlso
```
Comment (DustinCampbell) [inline src/Workspaces/VisualBasic/Portable/Formatting/Rules/AdjustSpaceFormattingRule.vb]: Spot the bug here.

Comment (Pilchie) [inline src/Workspaces/VisualBasic/Portable/Formatting/Rules/AdjustSpaceFormattingRule.vb]: Token kinds are not Expressions, they are Keywords :smile:

Comment (DustinCampbell) [inline src/Workspaces/VisualBasic/Portable/Formatting/Rules/AdjustSpaceFormattingRule.vb]: And note how the test below never caught it. :smile:

**Comment by DustinCampbell** [comment]
@balajikris, I added another test for chained conditional access expressions.

**Comment by balajikris** [comment]
:shipit:

**Comment by DustinCampbell** [comment]
retest this please

**Comment by DustinCampbell** [comment]
retest this please

**Comment by DustinCampbell** [comment]
Linux build failed due to Nuget package issues that are unrelated to this change. Merging.

**Pull request merged**